skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/toggleBulletList.ts

```diff
@@ -0,0 +1,436 @@
+  const theIdexOfTheContent = literalIndex.contentIndex;
+  const sentence = getSurroundingLiteralsAndVariables(thisBlock.content, theIdexOfTheContent);
+  const sentenceElements = sentence.map((r) => r.element);
+
+  const replacedWithItemList = replaceElementsBetweenIncluding(
+    thisBlock.content,
+    sentence[0].originalIndex,
+    sentence.at(-1)!.originalIndex,
+    newItemList({
+      items: [{ id: null, content: sentenceElements }],
+    }),
+  );
+
+  const { newcontent, newDeleted } = mergeItemLists(replacedWithItemList);
+
+  const newThisParagraph: ParagraphBlock = newParagraph({
+    id: thisBlock.id,
+    content: newcontent,
+    deletedContent: [...sentenceElements.filter((s) => !!s.id).map((r) => r.id!), ...newDeleted],
+  });
+
+  draft.redigertBrev.blocks[literalIndex.blockIndex] = newThisParagraph;
+  //---------------------------------------------
+
+  //---- Merging av punktlister dersom det finnes en punktliste-blokk før ----------------
+  const doesPreviousBlockEndWithAnItemList =
+    draft.redigertBrev.blocks[literalIndex.blockIndex - 1]?.content.at(-1)?.type === "ITEM_LIST";
+
+  const doesThisBlockStartWithAnItemList = newThisParagraph.content[0].type === "ITEM_LIST";
+
+  if (doesPreviousBlockEndWithAnItemList && doesThisBlockStartWithAnItemList) {
+    const previousBlock = draft.redigertBrev.blocks[literalIndex.blockIndex - 1];
+
+    const previousBlockContentBeforeItemList = previousBlock.content.slice(0, -1);
+    const previousBlockContentItemList = previousBlock.content.at(-1) as ItemList;
+
+    const thisBlockContentItemList = newThisParagraph.content[0] as ItemList;
+    const thisBlockContentAfterItemList = newThisParagraph.content.slice(1);
+
+    const thisBlockMergedWithPreviousBlock = newParagraph({
+      id: previousBlock.id,
+      content: [
+        ...previousBlockContentBeforeItemList,
+        newItemList({
+          items: [...previousBlockContentItemList.items, ...thisBlockContentItemList.items],
+          deletedItems: [...previousBlockContentItemList.deletedItems, ...thisBlockContentItemList.deletedItems],
+        }),
+        ...thisBlockContentAfterItemList,
+      ],
+      deletedContent: [...previousBlock.deletedContent, ...newThisParagraph.deletedContent],
+    });
+
+    const newBlocksWhereMergedWithPrevious = [
+      ...draft.redigertBrev.blocks.slice(0, literalIndex.blockIndex - 1),
+      thisBlockMergedWithPreviousBlock,
+      ...draft.redigertBrev.blocks.slice(literalIndex.blockIndex + 1),
+    ];
+
+    draft.redigertBrev.blocks = newBlocksWhereMergedWithPrevious;
+    draft.redigertBrev.deletedBlocks = [
+      ...draft.redigertBrev.deletedBlocks,
+      ...(newThisParagraph.id ? [newThisParagraph.id] : []),
+    ];
+  }
+  //---------------------------------------------
+
+  //her tar vi høyde for at det kan potensielt ha skjedd en merge eller ikke
+  const possibleNewThisBlockIndex = draft.redigertBrev.blocks.findIndex((block) =>
```

Comment:
Dette kan kanskje være en litt sketchy måte å identifisere det på? Sannsynligheten for at feil blokk blir identifsert er nok lav, men det kan potensielt lede til en obskur feil.